block/src/async_io.rs

```diff
@@ -82,6 +82,12 @@ pub enum AsyncIoError {
     /// Failed submitting batch requests.
     #[error("Failed submitting batch requests")]
     SubmitBatchRequests(#[source] std::io::Error),
+    /// Failed to punch hole file.
+    #[error("Failed to punch hole file: {0}")]
+    PunchHole(#[source] std::io::Error),
+    /// Failed to write zeroes to file.
+    #[error("Failed to write zeroes to file: {0}")]
+    WriteZeroes(#[source] std::io::Error),
```

Comment:
The nested error printing was improved a while ago and the `{0}` are not needed anymore with errors that provide a `#[source]`.
Also, a small suggestion:

```suggestion
    /// Failed to punch hole in file.
    #[error("Failed to punch hole in file")]
    PunchHole(#[source] std::io::Error),
    /// Failed to write zeroes to file.
    #[error("Failed to write zeroes to file")]
    WriteZeroes(#[source] std::io::Error),
```